scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -376,32 +366,31 @@ private void loggingEligibleTasksDetails(LinkedList<EligibleTaskDescriptor> full
      * The check of compliance is currently done by the {@link SchedulingTaskComparator} class.<br>
      * This method has two side effects : extracted tasks are removed from the bagOfTasks and put in the toFill list
      *
-     * @param bagOfTasks the list of tasks form which to extract tasks
+     * @param bagOfTasks  the list of tasks form which to extract tasks
      * @param maxResource the limit number of resources that the extraction should not exceed
-     * @param toFill the list that will contains the task to schedule at the end. This list must not be null but must be empty.<br>
-     * 		  this list will be filled with the n first compatible tasks according that the number of resources needed
-     * 		  by these tasks does not exceed the given max resource number.
+     * @param toFill      the list that will contains the task to schedule at the end. This list must not be null but must be empty.<br>
+     *                    this list will be filled with the n first compatible tasks according that the number of resources needed
+     *                    by these tasks does not exceed the given max resource number.
      * @return the number of nodes needed to start every task present in the 'toFill' argument at the end of the method.
      */
-    protected int getNextcompatibleTasks(Map<JobId, JobDescriptor> jobsMap,
-            LinkedList<EligibleTaskDescriptor> bagOfTasks, int maxResource, LinkedList<EligibleTaskDescriptor> toFill) {
+    protected int getNextCompatibleTasks(Map<JobId, JobDescriptor> jobsMap, List<EligibleTaskDescriptor> bagOfTasks,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 30 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)